src/System.Collections.NonGeneric/tests/QueueTests.cs

```diff
@@ -479,6 +479,22 @@ public static void GetEnumerator(int count)
         }
 
         [Fact]
+        public static void GetEnumerator_IClonable_Clone()
+        {
+            Queue queue = Helpers.CreateIntQueue(10);
+
+            IEnumerator enumerator = queue.GetEnumerator();
+            ICloneable cloneableEnumerator = (ICloneable)enumerator;
+
+            IEnumerator clonedEnumerator = (IEnumerator)cloneableEnumerator.Clone();
+            Assert.NotSame(enumerator, clonedEnumerator);
+
+            // Cloned and original enumerators should enumerate separately.
+            while (enumerator.MoveNext()) ;
+            Assert.True(clonedEnumerator.MoveNext());
```

Comment:
It'd be good to verify as well that:
a) the sequence matches between the original and cloned enumerators
b) that the cloned enumerator starts at the point in the sequence where the clone was done, not necessarily at the start